uniflow-test/src/test/kotlin/io/uniflow/test/impl/AbstractSampleFlow.kt

```diff
@@ -25,6 +21,28 @@ abstract class AbstractSampleFlow(defaultState: UIState) : DataFlow, UIDataPubli
     val states = arrayListOf<UIState>()
     val events = arrayListOf<UIEvent>()
 
+    fun hasStates(vararg states: UIState) {
+        assert(this.states == states.toList()) { "should have states ${states.toList()} but was ${this.states}" }
+    }
+
+    fun hasState(index: Int, state: UIState) {
```

Comment:
Note the difference in the API between this class and the SimpleTestObservers: there, the argument order is the other way around.